src/Client.js

```diff
@@ -1037,13 +1037,9 @@ class Client extends EventEmitter {
      * @returns {Promise<Object|null>}
      */
     async getNumberId(number) {
-        if (!number.endsWith('@c.us')) {
-            number += '@c.us';
-        }
-
         return await this.pupPage.evaluate(async number => {
-            const wid = window.Store.WidFactory.createWid(number);
-            const result = await window.Store.QueryExist(wid);
+            if (number.endsWith('@c.us')) number.replace('@c.us','');
+            const result = await window.Store.QueryExist({type: 'phone', phone: number.startsWith('+') ? number : `+${number}`});
```

Comment:
This line is incorrect. Instead of sending an object as a param with properties `type` and `phone`, WhatsApp internals are expecting 2 parameters here to work as expected instead an object. Changing this line to the following should fix the problem:

const result = await window.Store.QueryExist('phone', number.startsWith('+') ? number : `+${number}`);

Reply:
Still Not Working. 

`user must have an id or a phone
    at t.USyncUser.validate (app.77ee8aa7a4c593101105.js:50:58574)
    at app.77ee8aa7a4c593101105.js:50:50786
    at Array.filter (<anonymous>)
    at t.USyncQuery._buildStanza (app.77ee8aa7a4c593101105.js:50:50718)
    at app.77ee8aa7a4c593101105.js:50:51431
    at Generator.next (<anonymous>)
    at t (vendor1~app.58a51d93a250f83ab555.js:2:66483)
    at s (vendor1~app.58a51d93a250f83ab555.js:2:66694)
    at vendor1~app.58a51d93a250f83ab555.js:2:66753
    at Y (app.77ee8aa7a4c593101105.js:40:119783)`

Reply:
<img width="899" alt="Screen Shot 2022-11-23 at 2 49 59 PM" src="https://user-images.githubusercontent.com/39593002/203615204-ac3b167b-4699-444d-9db4-de75593e424c.png">

Yes, it works. Make sure you are not using a cached version of npm modules

Reply:
> This line is incorrect. Instead of sending an object as a param with properties `type` and `phone`, WhatsApp internals are expecting 2 parameters here to work as expected instead an object. Changing this line to the following should fix the problem:
> 
> ```
> const result = await window.Store.QueryExist('phone', number.startsWith('+') ? number : `+${number}`);
> ```

Depending on your region, the queryExist function receives parameters in different ways. An update has been made to deal with this.